audb/core/dependencies.py

```diff
@@ -375,52 +372,79 @@ def _add_media(
+                    if bit_depth is None:  # pragma: nocover (non SND files)
+                        bit_depth = 0
+                    channels = audiofile.channels(path)
+                    duration = audiofile.duration(path, sloppy=True)
+                    sampling_rate = audiofile.sampling_rate(path)
+                except FileNotFoundError:  # pragma: nocover
+                    # If sox or mediafile are not installed
+                    # we get a FileNotFoundError error
+                    raise RuntimeError(
+                        f"sox and mediainfo have to be installed "
+                        f"to publish '{format}' media files."
+                    )
+
+            row = (
+                archive,
+                bit_depth,
+                channels,
+                checksum,
+                duration,
+                format,
+                0,  # removed
+                sampling_rate,
+                define.DependType.MEDIA,
+                version,
+            )
+            rows.append(row)
+            files.append(file)
 
-        if archive is None:
-            archive = self.archive(file)
-
-        if checksum is None:
-            checksum = self.checksum(file)
-            bit_depth = self.bit_depth(file)
-            channels = self.channels(file)
-            duration = self.duration(file)
-            sampling_rate = self.sampling_rate(file)
-        else:
-            try:
-                path = os.path.join(root, file)
-                bit_depth = audiofile.bit_depth(path)
-                if bit_depth is None:  # pragma: nocover (non SND files)
-                    bit_depth = 0
-                channels = audiofile.channels(path)
-                duration = audiofile.duration(path, sloppy=True)
-                sampling_rate = audiofile.sampling_rate(path)
-            except FileNotFoundError:  # pragma: nocover
-                # If sox or mediafile are not installed
-                # we get a FileNotFoundError error
-                raise RuntimeError(
-                    f"sox and mediainfo have to be installed "
-                    f"to publish '{format}' media files."
-                )
+        df = pd.DataFrame.from_records(
+            rows,
+            index=files,
+            columns=define.DEPEND_FIELD_NAMES.values(),
+        )
 
-        self._df.loc[file] = [
-            archive,
-            bit_depth,
-            channels,
-            checksum,
-            duration,
-            format,
-            0,  # removed
-            sampling_rate,
-            define.DependType.MEDIA,
-            version,
-        ]
+        mask = df.index.isin(self._df.index)
+
+        # unfortunately, we cannot use update()
+        # as is does not preserve dtypes at the moment:
+        # https://github.com/pandas-dev/pandas/issues/4094
+        # self._df.update(df[mask])
+        self._df[self._df.index.isin(df.index)] = df[mask]
```

Comment:
The following works as well, but I'm not sure yet which is faster ;)

```suggestion
        self._df.loc[df.index[mask]] = df[mask]
```

Reply:
Yes, let's see. If we introduce an `update_media` function we could further simplify to: `self._df.loc[df.index] = df`.

Reply:
Good point, so I think it makes more sense to first tackle https://github.com/audeering/audb/issues/141 and then see, what is the fastest way to address the dataframe.